--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\logging\__init__.py", line 1103, in emit
    stream.write(msg + self.terminator)
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f680' in position 33: character maps to <undefined>
Call stack:
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\shael\Desktop\Automation Testing\shael\Scripts\pytest.exe\__main__.py", line 7, in <module>
    sys.exit(console_main())
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 201, in console_main
    code = main()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 336, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 289, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 343, in _main
    config.hook.pytest_runtestloop(session=session)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 367, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 117, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 136, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 245, in call_and_report
    call = CallInfo.from_call(
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 344, in from_call
    result: TResult | None = func()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "C:\Users\shael\Desktop\Automation Testing\test_scripts\day18\test_duckduckgo_search.py", line 11, in test_duckduckgo_search
    logger.info(f"🚀 Starting test with search term: {search_term}")
Message: '🚀 Starting test with search term: Automation Testing'
Arguments: ()
2025-09-20 15:02:03,359 - INFO - 🚀 Starting test with search term: Automation Testing
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\logging\__init__.py", line 1103, in emit
    stream.write(msg + self.terminator)
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f310' in position 33: character maps to <undefined>
Call stack:
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\shael\Desktop\Automation Testing\shael\Scripts\pytest.exe\__main__.py", line 7, in <module>
    sys.exit(console_main())
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 201, in console_main
    code = main()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 336, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 289, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 343, in _main
    config.hook.pytest_runtestloop(session=session)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 367, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 117, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 136, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 245, in call_and_report
    call = CallInfo.from_call(
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 344, in from_call
    result: TResult | None = func()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "C:\Users\shael\Desktop\Automation Testing\test_scripts\day18\test_duckduckgo_search.py", line 15, in test_duckduckgo_search
    logger.info("🌐 Opened DuckDuckGo homepage")
Message: '🌐 Opened DuckDuckGo homepage'
Arguments: ()
2025-09-20 15:02:04,334 - INFO - 🌐 Opened DuckDuckGo homepage
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\logging\__init__.py", line 1103, in emit
    stream.write(msg + self.terminator)
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f50d' in position 33: character maps to <undefined>
Call stack:
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\shael\Desktop\Automation Testing\shael\Scripts\pytest.exe\__main__.py", line 7, in <module>
    sys.exit(console_main())
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 201, in console_main
    code = main()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 336, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 289, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 343, in _main
    config.hook.pytest_runtestloop(session=session)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 367, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 117, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 136, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 245, in call_and_report
    call = CallInfo.from_call(
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 344, in from_call
    result: TResult | None = func()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "C:\Users\shael\Desktop\Automation Testing\test_scripts\day18\test_duckduckgo_search.py", line 18, in test_duckduckgo_search
    logger.info(f"🔍 Searching for: {search_term}")
Message: '🔍 Searching for: Automation Testing'
Arguments: ()
2025-09-20 15:02:05,261 - INFO - 🔍 Searching for: Automation Testing
2025-09-20 15:02:07,134 - INFO - First result text: Automation Testing - Software Testing - GeeksforGeeks
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\logging\__init__.py", line 1103, in emit
    stream.write(msg + self.terminator)
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2705' in position 33: character maps to <undefined>
Call stack:
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\shael\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\shael\Desktop\Automation Testing\shael\Scripts\pytest.exe\__main__.py", line 7, in <module>
    sys.exit(console_main())
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 201, in console_main
    code = main()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\config\__init__.py", line 175, in main
    ret: ExitCode | int = config.hook.pytest_cmdline_main(config=config)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 336, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 289, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 343, in _main
    config.hook.pytest_runtestloop(session=session)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\main.py", line 367, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 117, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 136, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 245, in call_and_report
    call = CallInfo.from_call(
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 344, in from_call
    result: TResult | None = func()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\pluggy\_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\shael\Desktop\Automation Testing\shael\lib\site-packages\_pytest\python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "C:\Users\shael\Desktop\Automation Testing\test_scripts\day18\test_duckduckgo_search.py", line 27, in test_duckduckgo_search
    logger.info(f"✅ Search successful for: {search_term}")
Message: '✅ Search successful for: Automation Testing'
Arguments: ()
2025-09-20 15:02:07,198 - INFO - ✅ Search successful for: Automation Testing
